src/web/event-notifications/event-notification-types/SlackNotificationForm.jsx

```diff
@@ -141,9 +174,31 @@ class SlackNotificationForm extends React.Component {
                  label="Custom Message (optional)"
                  type="textarea"
                  bsStyle={validation.errors.custom_message ? 'error' : null}
-                 help={get(validation, 'errors.custom_message[0]', 'Custom message to be appended below the alert title. See https://docs.graylog.org/en/latest/pages/alerts.html#data-available-to-notifications for more details.')}
+                 help={get(validation, 'errors.custom_message[0]', element)}
                  value={config.custom_message || ''}
                  onChange={this.handleChange} />
+
+          <FormGroup>
+            <ControlLabel>Backlog Message Items (optional)</ControlLabel>
+            <InputGroup>
+              <InputGroup.Addon>
+                <input id="toggle_backlog_size"
+                       type="checkbox"
+                       checked={isBacklogSizeEnabled}
+                       onChange={this.toggleBacklogSize} />
+              </InputGroup.Addon>
+              <FormControl type="number"
+                           id="backlog_size"
+                           name="backlog_size"
+                           onChange={this.handleBacklogSizeChange}
+                           value={backlogSize}
+                           min="0"
```

Comment:
Since users seem to be able to turn this on and off with a checkbox, I am concerned about the behavior when they turn it on and set the value to zero.  

As a user, if I turn on the limit and set the limit to 0, I would expect to see my backlog messages limited to zero rather than unlimited.

Perhaps we should be using -1 as the flag value rather than zero?  Or we should make the minimum (when active) 1?  It just seems like we are creating a potentially confusing user experience.

Reply:
the behavior is well documented in `test_backlog_messagesize`method and applying this change, will lead to a breaking unit tests. if you have further concerns, please create a bug and we can address this in a seperate issue. 

 //global backlog messages is 0 ,when the SlackEventNotificationConfig backlog_size is set to 0, then the expected backlog message size is 0.

Reply:
Comments in a JUnit are not the same as user-facing documentation.  Here is what I see when I test this locally:

* I open the configuration for my Slack notification
* I check the box for message backlog limit
* I set the value to 0 - as a User, I expect that I will receive 0 backlog messages in my notification
* I save my changes
* When the next notification happens, I see unlimited backlog messages in the notification
* I re-open the configuration for my Slack notification
* I see that the message backlog limit checkbox is unchecked

Reply:
if i understand right, can you please rewrite your thoughts like below, just so we stay consistent in our language, 
 
//global backlog message is 10,when the SlackEventNotificationConfig backlog_size is set to 0, then the expected backlog message size is 10.